test/collection.js

```diff
@@ -656,6 +656,34 @@ describe('Chained Collection', () => {
     ).to.reject()
   })
 
+  it('reverse', async () => {
+    expect(
+      await Collect([1, 2, 3]).reverse().all()
+    ).to.equal([3, 2, 1])
+
+    expect(
+      await Collect([1]).reverse().all()
+    ).to.equal([1])
+
+    expect(
+      await Collect([]).reverse().all()
+    ).to.equal([])
+
+    expect(
+      await Collect([1, 2, 3, 2, 1]).reverse().all()
+    ).to.equal([1, 2, 3, 2, 1])
+
+    const items = [1, 2, 3]
+    const collection = Collect(items)
+
+    expect(
+      await collection.reverse().all()
+    ).to.equal([3, 2, 1])
+    expect(
+      await collection.reverse().all()
```

Comment:
Can you please remove the `.reverse()` call in this chain? I'd love to test that the original collection isn't affected by the reverse operation and stays in the same order.

```suggestion
      await collection.all()
```